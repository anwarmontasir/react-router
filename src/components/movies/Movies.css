:local(.movies) {
  display: flex;
  flex-wrap: wrap;
}