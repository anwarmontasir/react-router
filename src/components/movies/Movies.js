import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Movie from './Movie';
import styles from './Movies.css';

export default class Movies extends Component {
  
  static propTypes = {
    movies: PropTypes.array
  };

  render() {
    const { movies } = this.props;

    return (
      <ul className={styles.movies}>
        {movies.map(movie=> <Movie key={movie.imdbID} {...movie}/>)}
      </ul>
    );
  }
}