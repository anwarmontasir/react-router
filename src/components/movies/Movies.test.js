import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Movies from './Movies';

describe('Movies', () => {
  it('renders movies', () => {
    const movies =  [{
      'Title': 'Star Wars: Episode IV - A New Hope',
      'Year': '1977',
      'imdbID': 'tt0076759',
      'Type': 'movie',
      'Poster': 'https://images-na.ssl-images-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg'
    }, {
      'Title': 'Star Wars: Episode V - The Empire Strikes Back',
      'Year': '1980',
      'imdbID': 'tt0080684',
      'Type': 'movie',
      'Poster': 'https://images-na.ssl-images-amazon.com/images/M/MV5BYmU1NDRjNDgtMzhiMi00NjZmLTg5NGItZDNiZjU5NTU4OTE0XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg'
    }];

    const wrapper = shallow(<Movies movies={movies}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});