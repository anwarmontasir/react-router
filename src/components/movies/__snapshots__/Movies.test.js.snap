// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies renders movies 1`] = `
<ul>
  <Movie
    Poster="https://images-na.ssl-images-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
    Title="Star Wars: Episode IV - A New Hope"
    Type="movie"
    Year="1977"
    imdbID="tt0076759"
    key="tt0076759"
  />
  <Movie
    Poster="https://images-na.ssl-images-amazon.com/images/M/MV5BYmU1NDRjNDgtMzhiMi00NjZmLTg5NGItZDNiZjU5NTU4OTE0XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
    Title="Star Wars: Episode V - The Empire Strikes Back"
    Type="movie"
    Year="1980"
    imdbID="tt0080684"
    key="tt0080684"
  />
</ul>
`;
